components/rdf-load.js

```diff
@@ -4,48 +4,58 @@ var _ = require('underscore');
 var Promise = require('promise');
 var rdfstore = require('rdfstore');
 
-var basenode = require('../src/base-node');
-var promiseComponent = require('../src/promise-component');
+var promiseOutput = require('../src/promise-output');
+var componentFactory = require('../src/noflo-component-factory');
 
-exports.getComponent = promiseComponent({
+exports.getComponent = componentFactory({
     description: "Loads data into a RDF JS Interface Graph object",
     icon: 'sign-in',
+    outPorts: promiseOutput.outPorts,
     inPorts: {
         options: {
             description: "A map of configuration options for the store",
             datatype: 'object',
-            ondata: basenode.assign('options')
+            ondata: function(options) {
+                this.nodeInstance.options = options;
+            }
         },
         media: {
             description: "Media type (application/json, text/n3...) of the data to be parsed or the value 'remote' if a URI for the data is passed instead",
             datatype: 'string',
-            ondata: basenode.assign('media')
+            ondata: function(media) {
+                this.nodeInstance.media = media;
+            }
         },
         graph: {
             description: "Graph URI template where the parsed triples will be inserted. If it is not specified, triples will be loaded in the default graph",
             datatype: 'string',
-            ondata: basenode.assign('graph')
+            ondata: function(graph) {
+                this.nodeInstance.graph = graph;
+            }
         },
-        'in': {
+        input: {
             description: "RDF data to be parsed and loaded or an URI where the data will be retrieved after performing content negotiation",
             datatype: 'all',
             required: true,
-            ondata: basenode.assign('data'),
-            ondisconnect: load
+            ondata: function(data) {
+                this.nodeInstance.data = data;
+            },
+            ondisconnect: promiseOutput(load)
         }
     }
 });
 
 function load() {
```

Comment:
I don't think the load function should be moved anywhere else. The term graph here refers to an RDF graph and the options object is for use with the rdfstore. Neither of which are about noflo.